dispatcher.go

```diff
@@ -52,20 +52,28 @@ type Dispatcher interface {
 	// OnDeviceEvent is the device.Listener function that processes outbound events.  Inject
 	// this function as a device listener for a manager.
 	OnDeviceEvent(*device.Event)
+	// OnDeviceEventQOSAck is the device.Listener function that processes outbound
+	// events and determines whether or not a QOS ack to the source device is required.
+	// Inject this function as a device listener for a manager.
+	OnDeviceEventQOSAck(*device.Event)
```

Comment:
My thinking was that we may need to do things differently depending on the `event.Type` in the future.
So making QOS ack as it's own listener felt right.

However, if we're only doing qos ack for `MessageReceived` event types then I'm down to converting it to a hidden method and calling it directly from `OnDeviceEvent`

Reply:
We will do things differently based on event type, but that shouldn't be expressed through a different method on the listener interface.  There's a few reasons for this:

(1) It breaks the ability for a `func(*device.Event)` type to implement `device.Listener`
(2) It breaks existing clients, since there's now a new, exported method that they won't be implementing
(3) It is inconsistent.  We don't have `OnMessageReceived`, `OnPing`, etc methods for all the same reasons listed here.
(4) Acks shouldn't be events.  Acks are something done in response to events.  Having a separate method makes that muddy.

The particular `OnDeviceEvent` implementation should just handle the ack.  And, it should probably be in a separate listener that *only* handles acks.  Talaria can add multiple listeners, which is by design.  That promotes a modular and configurable way have handling device events.